### ISSUE TITLE
"Content is Unreadable" error when strings contain control characters

### ISSUE THREAD

**Issue by sw1020** [jmcnamara/libxlsxwriter#42]
Issue with Content is Unreadable. Open and Repair

I am using libxlsxwriter to make a node addon.

I found : when I write the utf-8 string with some special characters , it will cause  the problem
for exzample:   <Buffer e5 9b be 14 e5 9b be>

pls help

**Comment by jmcnamara** [comment]
The "Content is Unreadable. Open and Repair" error is Excel's standard warning for any validation error in the XML used for the components of the XLSX file. It doesn't really help to identify the problem.

You will need to submit a small working example that demonstrates the issue. See the [Bug Reporting Guidelines](https://github.com/jmcnamara/libxlsxwriter/blob/master/CONTRIBUTING.md).

Note, libxlsxwriter does work with UTF-8. There is an [example in the docs/repo](http://libxlsxwriter.github.io/utf8_8c-example.html) and there are several test cases that test it.

John

**Comment by sw1020** [comment]
sorry 
the utf-8 string example (in hex): e5 9b be 14 e5 9b be
if i replace 0x14 to 0x20 ,it will work fine

**Comment by jmcnamara** [comment]
The non-unicode character 0x14 is a control character. These aren't currently supported in libxlsxwriter.

I'll fix that in a later version but for now make sure that your strings don't contain control characters.

**Comment by jmcnamara** [comment]
Here is an example program that demonstrates the issue:

```c
#include "xlsxwriter.h"

int main() {

    lxw_workbook  *workbook  = new_workbook("gh41_1.xlsx");
    lxw_worksheet *worksheet = workbook_add_worksheet(workbook, NULL);

    char string[] = {0xe5, 0x9b, 0xbe, 0x14, 0xe5, 0x9b, 0xbe};
    
    worksheet_write_string(worksheet, 0, 0, string, NULL);

    workbook_close(workbook);

    return 0;
}
```
And here is a working example with the control character replaced with a space:

```c
#include "xlsxwriter.h"

int main() {

    lxw_workbook  *workbook  = new_workbook("gh41_2.xlsx");
    lxw_worksheet *worksheet = workbook_add_worksheet(workbook, NULL);

    char string[] = {0xe5, 0x9b, 0xbe, 0x20, 0xe5, 0x9b, 0xbe};
    
    worksheet_write_string(worksheet, 0, 0, string, NULL);

    workbook_close(workbook);

    return 0;
}
```

**Comment by jmcnamara** [comment]
Fixed in version 0.2.7 and the master HEAD on Github.

Thanks.
